src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,96 @@
+    private static final String INDEX_NAME_DELIMITER = ":";
+    private static final String INDEX_NAME_PAIRS_DELIMITER = "|";
+    private static final String REPORT_DELIMITER = " - ";
+    private static final String REPORT_HEADER = "Report for period ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String indexedNameString = buildIndexedNameString(names);
+        int[] salaries = new int[names.length];
+        for (String dataRecord : data) {
+            String[] records = dataRecord.split(DATA_DELIMITER);
+            if (!indexedNameString.contains(records[NAME_INDEX])) {
+                continue;
+            }
+            LocalDate parsedRecordDate = parseStringToLocalDate(records[DATE_INDEX]);
+            if (parsedRecordDate.isBefore(parseStringToLocalDate(dateFrom))) {
+                continue;
+            }
+            if (parsedRecordDate.isAfter(parseStringToLocalDate(dateTo))) {
+                continue;
+            }
+            int salariesIndex = getSalariesIndexByName(indexedNameString, records[NAME_INDEX]);
+            try {
+                salaries[salariesIndex] +=
+                        Integer.parseInt(records[INCOME_INDEX]) * Integer.parseInt(
+                                records[HOURS_INDEX]);
+            } catch (NumberFormatException e) {
+                throw new RuntimeException(
+                        "Wrong format of working hours or income per hour in " + "records line.");
+            }
+        }
+        return composeReport(names, dateFrom, dateTo, salaries);
+    }
+
+    private static int getSalariesIndexByName(String indexedNameString, String employeeName) {
+        int startIndex = indexedNameString.lastIndexOf(INDEX_NAME_PAIRS_DELIMITER,
+                indexedNameString.indexOf(employeeName));
+        int endIndex = indexedNameString.indexOf(INDEX_NAME_DELIMITER, startIndex);
+        String indexSalaries = indexedNameString.substring(startIndex + 1, endIndex);
+        try {
+            return Integer.parseInt(indexSalaries);
+        } catch (NumberFormatException e) {
+            throw new RuntimeException("Wrong index format in indexed string of employee names.");
+        }
+    }
+
+    private static String composeReport(String[] names, String dateFrom, String dateTo,
+                                        int[] salaries) {
+        StringBuilder reportBuilder = new StringBuilder(REPORT_HEADER).append(dateFrom)
+                .append(REPORT_DELIMITER)
+                .append(dateTo)
+                .append(System.lineSeparator());
+        for (int i = 0; i < names.length; i++) {
+            reportBuilder.append(names[i]).append(" - ").append(salaries[i]);
+            if (i < names.length - 1) {
+                reportBuilder.append(System.lineSeparator());
+            }
+        }
+        return String.valueOf(reportBuilder);
```

Comment:
There is no need to use 'String.valueOf(reportBuilder)' since 'reportBuilder' is already a StringBuilder and calling 'toString()' on it will give you the string representation. You can simply return 'reportBuilder.toString()'.